central/telemetry/phonehome/interceptor.go

```diff
@@ -0,0 +1,97 @@
+package phonehome
+
+import (
+	"context"
+	"net/http"
+	"strings"
+
+	"github.com/pkg/errors"
+	erroxGRPC "github.com/stackrox/rox/pkg/errox/grpc"
+	"github.com/stackrox/rox/pkg/grpc/authn"
+	grpcError "github.com/stackrox/rox/pkg/grpc/errors"
+	"github.com/stackrox/rox/pkg/grpc/requestinfo"
+	"github.com/stackrox/rox/pkg/httputil"
+	"github.com/stackrox/rox/pkg/set"
+	pkgPH "github.com/stackrox/rox/pkg/telemetry/phonehome"
+	"google.golang.org/grpc"
+)
+
+var (
+	ignoredPaths = []string{"/v1/ping", "/v1/metadata", "/static/"}
+)
+
+func track(ctx context.Context, t pkgPH.Telemeter, err error, info *grpc.UnaryServerInfo, trackedPaths set.FrozenSet[string]) {
+	userAgent, userID, path, code := getRequestDetails(ctx, err, info)
+
+	// Track the API path and error code of some requests:
+
+	for _, ip := range ignoredPaths {
+		if strings.HasPrefix(path, ip) {
+			return
+		}
+	}
+
+	if trackedPaths.Contains("*") || trackedPaths.Contains(path) {
+		t.Track("API Call", userID, map[string]any{
+			"Path":       path,
+			"Code":       code,
+			"User-Agent": userAgent,
+		})
+	}
+}
+
+func getRequestDetails(ctx context.Context, err error, info *grpc.UnaryServerInfo) (userAgent string, userID string, method string, code int) {
+	ri := requestinfo.FromContext(ctx)
+	userAgent = strings.Join(ri.Metadata.Get("User-Agent"), ", ")
+
+	id, iderr := authn.IdentityFromContext(ctx)
+	if iderr != nil {
+		userID = "unauthenticated"
+		log.Debug("Cannot identify user from context: ", iderr)
+	} else {
+		userID = pkgPH.HashUserID(id.UID())
+	}
+
+	if ri.HTTPRequest != nil && ri.HTTPRequest.URL != nil {
+		method = ri.HTTPRequest.URL.Path
+		code = grpcError.ErrToHTTPStatus(err)
+	} else if info != nil {
+		method = info.FullMethod
+		code = int(erroxGRPC.RoxErrorToGRPCCode(err))
+	} else {
+		method = "unknown"
+		code = -1
```

Comment:
Might be good to have a comment explaining in which scenarios we would fall into this (or just simply that we do not expect this to be the case at all but have it for safety reasons).